config.go

```diff
@@ -146,6 +186,58 @@ func (u *UserSettings) Get(alias, key string) string {
 // error will be non-nil if and only if a user's configuration file or the
 // system configuration file could not be parsed, and u.IgnoreErrors is false.
 func (u *UserSettings) GetStrict(alias, key string) (string, error) {
+	u.doLoadConfigs()
+	if u.onceErr != nil && u.IgnoreErrors == false {
+		return "", u.onceErr
+	}
+	val, err := findVal(u.userConfig, alias, key)
+	if err != nil || val != "" {
+		return val, err
+	}
+	val2, err2 := findVal(u.systemConfig, alias, key)
+	if err2 != nil || val2 != "" {
+		return val2, err2
+	}
+	return Default(key), nil
+}
+
+// GetAllStrict retrieves zero or more directives for key for the given alias.
+// If key has a default value and no matching configuration is found, the
+// default will be returned. For more information on default values and the way
+// patterns are matched, see the manpage for ssh_config.
+//
+// error will be non-nil if and only if a user's configuration file or the
+// system configuration file could not be parsed, and u.IgnoreErrors is false.
+func (u *UserSettings) GetAllStrict(alias, key string) ([]string, error) {
+	u.doLoadConfigs()
+	if u.onceErr != nil && u.IgnoreErrors == false {
+		return nil, u.onceErr
+	}
+	val, err := findAll(u.userConfig, alias, key)
+	if err != nil || val != nil {
+		return val, err
+	}
+	val2, err2 := findAll(u.systemConfig, alias, key)
+	if err2 != nil || val2 != nil {
+		return val2, err2
+	}
+
+	if strings.ToLower(key) == "identityfile" {
+		switch u.Get(alias, "Protocol") {
+		case "1":
+			return []string{defaults["identityfile"]}, nil
+		case "2":
+			def := make([]string, len(defaultProtocol2Identities))
+			copy(def, defaultProtocol2Identities)
+			return def, nil
+		}
+	}
+
+	// currently, non-identityfile directives don't have defaults
```

Comment:
I'm confused about this. If you ask for GetStrict("localhost", "Port") and we return you 22 if there's no match, it seems like we should return you `[]string{"22"}` from `GetAllStrict("localhost", "Port")` if there's no match.